x11-packages/wlroots/0005-backend-termuxgui.patch

```diff
@@ -0,0 +1,1799 @@
+--- a/backend/backend.c
++++ b/backend/backend.c
+@@ -36,6 +36,10 @@
+ #include <wlr/backend/x11.h>
+ #endif
+ 
++#if defined (__ANDROID__) && defined (__TERMUX__)
++#include <wlr/backend/termuxgui.h>
++#endif
++
+ #define WAIT_SESSION_TIMEOUT 10000 // ms
+ 
+ void wlr_backend_init(struct wlr_backend *backend,
+@@ -196,6 +200,23 @@ static struct wlr_backend *attempt_headless_backend(
+ 	return backend;
+ }
+ 
++#if defined (__ANDROID__) && defined (__TERMUX__)
++static struct wlr_backend *attempt_tgui_backend(
++		struct wl_display *display) {
++	struct wlr_backend *backend = wlr_tgui_backend_create(display);
++	if (backend == NULL) {
++		return NULL;
++	}
++
++	size_t outputs = parse_outputs_env("WLR_TGUI_OUTPUTS");
++	for (size_t i = 0; i < outputs; ++i) {
++		wlr_tgui_add_output(backend);
++	}
++
++	return backend;
++}
++#endif
++
+ #if WLR_HAS_DRM_BACKEND
+ static struct wlr_backend *attempt_drm_backend(struct wl_display *display,
+ 		struct wlr_backend *backend, struct wlr_session *session) {
+@@ -249,6 +270,10 @@ static bool attempt_backend_by_name(struct wl_display *display,
+ #endif
+ 	} else if (strcmp(name, "headless") == 0) {
+ 		backend = attempt_headless_backend(display);
++#if defined (__ANDROID__) && defined (__TERMUX__)
++	} else if (strcmp(name, "tgui") == 0) {
++		backend = attempt_tgui_backend(display);
++#endif
+ 	} else if (strcmp(name, "drm") == 0 || strcmp(name, "libinput") == 0) {
+ 		// DRM and libinput need a session
+ 		if (*session_ptr == NULL) {
+@@ -344,6 +369,18 @@ struct wlr_backend *wlr_backend_autocreate(struct wl_display *display,
+ 	}
+ #endif
+ 
++#if defined (__ANDROID__) && defined (__TERMUX__)
++	{
++		struct wlr_backend *tgui_backend = attempt_tgui_backend(display);
++		if (!tgui_backend) {
++			goto error;
++		}
++
++		wlr_multi_backend_add(multi, tgui_backend);
++		goto success;
++	}
++#endif
++
+ #if !(WLR_HAS_LIBINPUT_BACKEND || WLR_HAS_DRM_BACKEND)
+ 	wlr_log(WLR_ERROR, "Neither DRM nor libinput backend support is compiled in");
+ 	goto error;
+diff --git a/backend/meson.build b/backend/meson.build
+index ed977d3b..446d894d 100644
+--- a/backend/meson.build
++++ b/backend/meson.build
+@@ -27,3 +27,4 @@ endforeach
+ subdir('multi')
+ subdir('wayland')
+ subdir('headless')
++subdir('termuxgui')
+diff --git a/backend/multi/backend.c b/backend/multi/backend.c
+index 49792ff9..b269a702 100644
+--- a/backend/multi/backend.c
++++ b/backend/multi/backend.c
+@@ -209,6 +209,13 @@ bool wlr_multi_backend_add(struct wlr_backend *_multi,
+ 	wl_signal_add(&backend->events.new_output, &sub->new_output);
+ 	sub->new_output.notify = new_output_reemit;
+ 
++#if defined (__ANDROID__) && defined (__TERMUX__)
++	if (multi->backend.allocator == NULL &&
++		backend->allocator != NULL) {
++		multi->backend.allocator = backend->allocator;
++	}
++#endif
++
+ 	wl_signal_emit_mutable(&multi->events.backend_add, backend);
+ 	return true;
+ }
+diff --git a/backend/termuxgui/.clang-format b/backend/termuxgui/.clang-format
```

Comment:
It would be better to put these source in a separate directory rather than managing them in a patch file. You can copy them to the source directory in `termux_step_post_get_source`.

See https://github.com/termux/termux-packages/blob/7fcba28bf0f74f15a3479c470bac11d534adc5dd/packages/pulseaudio/module-aaudio-sink.c.